### ISSUE TITLE
tinker with shadcn

### ISSUE THREAD

**Issue by technophile-04** [scaffold-eth/scaffold-eth-2#935]
### Goal: 

Tinker with SE-2 x ShadCn to see if it nicely integrates with our needs.

We can go with 2 routes: 
- In base SE-2 replace daisyUI with shadcn
- we keep base SE-2 with daisyUI but allow shadcn through extensions

The rationale behind replacing daisUI with shadcn is because on surface level I think shadcn should be able to replace daisyUI in SE-2 but sometimes as we go deep into rabbit hole we hit some bad blockers that does not go with ethos of SE-2 like #444...but yeah lets tinker and see if route 1 is an option

Even if we don't go with 1st route, I think it's really good to have shadcn as core extension in [create-eth](https://github.com/scaffold-eth/create-eth)

**Comment by rin-st** [comment]
Great!!

> we keep base SE-2 with daisyUI but allow shadcn through extensions

I think it's not the case
- some shadcn components conflicts with daisyui, for example [sonner](https://ui.shadcn.com/docs/components/sonner). I tried a week ago and it didn't work without disabling `daisyui` plugin
- how should it work? We can't just replace daisyUI styles and logic inside all the core components and pages with shadcn. If not replace, everything will be the same daisyUI and just shadcnUi components will be added, so I can use them for creating my new components? I think it's not the case and no one will use it because all the app is still daisyUi
- (less important) tailwind config with both daisy and shadcn is starting to become a mess

> The rationale behind replacing daisUI with shadcn is because on surface level I think shadcn should be able to replace daisyUI in SE-2

Agree! 

Btw, shadcn works fine with `yarn`, but their yarn copy button uses npx for yarn (looks like a bug). I used `yarn dlx` instead
<img width="906" alt="image" src="https://github.com/user-attachments/assets/801fd8ee-1a05-4765-bbbc-961dbe0229b0">
